run_csm.py

```diff
@@ -0,0 +1,134 @@
+                raise
+
+def load_prompt_audio(audio_path: str, target_sample_rate: int) -> torch.Tensor:
+    audio_tensor, sample_rate = torchaudio.load(audio_path)
+    audio_tensor = audio_tensor.squeeze(0)
+    # Resample is lazy so we can always call it
+    audio_tensor = torchaudio.functional.resample(
+        audio_tensor, orig_freq=sample_rate, new_freq=target_sample_rate
+    )
+    return audio_tensor
+
+def prepare_prompt(text: str, speaker: int, audio_path: str, sample_rate: int) -> Segment:
+    audio_tensor = load_prompt_audio(audio_path, sample_rate)
+    return Segment(text=text, speaker=speaker, audio=audio_tensor)
+
+def main():
+    # Download prompts if needed
+    ensure_prompts_downloaded()
+
+    # Select the best available device, skipping MPS due to float64 limitations
+    if torch.cuda.is_available():
+        device = "cuda"
+    else:
+        device = "cpu"
+    print(f"Using device: {device}")
+    
+    try:
+        # Load model
+        generator = load_csm_1b(device)
+        
+        # Prepare prompts
+        prompt_a = prepare_prompt(
+            SPEAKER_PROMPTS["conversational_a"]["text"],
+            0,
+            SPEAKER_PROMPTS["conversational_a"]["audio"],
+            generator.sample_rate
+        )
+        
+        prompt_b = prepare_prompt(
+            SPEAKER_PROMPTS["conversational_b"]["text"],
+            1,
+            SPEAKER_PROMPTS["conversational_b"]["audio"],
+            generator.sample_rate
+        )
+        
+        # Generate conversation
+        conversation = [
+            {"text": "Hey how are you doing?", "speaker_id": 0},
+            {"text": "Pretty good, pretty good. How about you?", "speaker_id": 1},
+            {"text": "I'm great! So happy to be speaking with you today.", "speaker_id": 0},
+            {"text": "Me too! This is some cool stuff, isn't it?", "speaker_id": 1}
+        ]
+        
+        # Generate each utterance
+        generated_segments = []
+        prompt_segments = [prompt_a, prompt_b]
+        
+        for utterance in conversation:
+            print(f"Generating: {utterance['text']}")
+            audio_tensor = generator.generate(
+                text=utterance['text'],
+                speaker=utterance['speaker_id'],
+                context=prompt_segments + generated_segments,
+                max_audio_length_ms=10_000,
+            )
+            generated_segments.append(Segment(text=utterance['text'], speaker=utterance['speaker_id'], audio=audio_tensor))
+
+        # Concatenate all generations
+        all_audio = torch.cat([seg.audio for seg in generated_segments], dim=0)
+        torchaudio.save(
+            "full_conversation.wav",
+            all_audio.unsqueeze(0).cpu(),
+            generator.sample_rate
+        )
+        print("Successfully generated full_conversation.wav")
+            
+    except Exception as e:
+        print(f"Error: {str(e)}")
+        return
```

Comment:
What errors did you run into that made you add this?

Reply:
Since this is a cli, my preference is to remove the try-except, a stacktrace is often more helpful